app/models/fact.rb

```diff
@@ -25,8 +31,16 @@ def take_checkpoint(author)
     return checkpoint
   end
 
+  def dependent_fact_defining_spaces
```

Comment:
`dependent_fact_defining_spaces` returns a list of spaces that depend on this fact that also define facts themselves, so I think `space_inputs` wouldn't make that much sense. Unless perhaps I misunderstood your suggested renaming here?